### PR TITLE
Change definition of the home directory on Windows

### LINKED ISSUE #57: Windows home directory
On Windows, R sets the `R_USER` environment variable to the user's `Documents` folder, and R uses this as the 'home' directory (for e.g. tilde-path expansion). This implies that by default:

```R
> path.expand("~")
[1] "C:/Users/kevin/Documents"
> fs::path_home()
C:/Users/kevin/Documents
> Sys.getenv("R_USER")
[1] "C:/Users/kevin/Documents"
```

This seems a bit counter-intuitive as this folder is not really the user's "home" directory. The associated logic R uses to choose this path is here and below:

https://github.com/wch/r-source/blob/9bec1ec8b544e077cbb35f190ad9c62a99876190/src/gnuwin32/shext.c#L26-L45

Would it make sense for `fs` to enforce the 'real' home directory (as set with e.g. `%USERPROFILE%` or `%HOMEDRIVE%%HOMEPATH%`)?

### PR DESCRIPTION
**Pull request by jimhester**
Rs definition of the home directory on Windows differs from that
typically used in other programming languages such as python and other
windows programs. This makes finding things like the desktop directory
more cumbersome on windows than it would be if R used the normal
definition.

Fixes #57

### CODE REVIEW

**Comment by jimhester** [comment]
@jennybc it would be good to have you look over the documentation to make sure it is clear and motivates the change adequately (you can also review the code if you want to of course).

File: NEWS.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 ## Breaking changes
 
+* `path_expand()` and `path_home()` now use `USERPROFILE` or
+  `HOMEDRIVE`/`HOMEPATH` as the user home directory on Windows. This differs
+  from the definition used in `path.expand()` but is consistent with
+  definitions from other programming environments such as python and rust. To
+  mimic R's (and previous) behavior use `Sys.setenv("R_FS_HOME" = Sys.getenv("R_USER"))`
+
```
Comment (jennybc) [inline NEWS.md]: You could also point out that this is also more compatible with Git (and ssh), both of which, by default, put their user-level files in `USERPROFILE`.

Either here or in the help.

File: R/path.R [hunk 2]
```diff
@@ -224,36 +213,52 @@ path_rel <- function(path, start = ".") {
   path_tidy(path)
 }
 
-#' Paths starting from useful directories
+#' Finding Users Home Directory
```
Comment (jennybc) [inline R/path.R]: User's? or User?

File: R/path.R [hunk 3]
```diff
@@ -224,36 +213,52 @@ path_rel <- function(path, start = ".") {
...
+#'
+#' * `path_expand()` performs tilde expansion on a path, replacing instances of
+#' `~` or `~user` with the user's home directory.
+#' * `path_home()` constructs a path within the expanded users home directory
```
Comment (jennybc) [inline R/path.R]: The utility of calling `path_home()` w/ no arguments to see what fs regards as home directory seems worth pointing out in one or more places.

File: R/path.R [hunk 4]
```diff
@@ -224,36 +213,52 @@ path_rel <- function(path, start = ".") {
...
+#' @details
+#' [path_expand()] Differs from [path.expand()] in the interpretation of the
```
Comment (jennybc) [inline R/path.R]: Why not `path_expand()` here?

File: R/path.R [hunk 5]
```diff
@@ -224,36 +213,52 @@ path_rel <- function(path, start = ".") {
...
-#' It first tries to use the Windows "personal" directory (typically
-#' `C:\\Users\\username\\Documents`). If that fails, if both environment
-#' variables `HOMEDRIVE` and `HOMEPATH` are set (and they normally are), the
-#' value is `${HOMEDRIVE}${HOMEPATH}`. If all of these fail, the current
-#' working directory is used. Source: [R for Windows FAQ - 2.14](https://cran.r-project.org/bin/windows/base/rw-FAQ.html#What-are-HOME-and-working-directories_003f).
+#' This definition makes fs more consistent with the definition of home directory used
+#' on Windows in other languages, such as
+#' [python](https://docs.python.org/3/library/os.path.html#os.path.expanduser)
+#' and [rust](https://doc.rust-lang.org/std/env/fn.home_dir.html#windows).
+#'
```
Comment (jennybc) [inline R/path.R]: Feels like this also makes it easier to write paths relative to "user's home directory" that are portable across Windows and *nix, esp. Mac OS. That's worth pointing out explicitly somewhere.

File: NEWS.md (hunk 1 above)
Comment (jennybc) [reply NEWS.md]: Summary of my research on this re: ssh keys:

https://github.com/ropensci/git2r/issues/314#issuecomment-359135124

File: R/path.R [hunk 6]
```diff
@@ -224,36 +213,52 @@ path_rel <- function(path, start = ".") {
...
+#' path_expand("~")
+#' }
 #' path_home()
 #' path_home("R")
-#'
-#' path_temp()
-#' path_temp("does-not-exist")
-path_home <- function(...) {
-  path(path_expand("~/"), ...)
+path_expand <- function(path) {
```
Comment (jennybc) [inline R/path.R]: Apparently this `path` param is not documented anywhere at this point.

Comment (jimhester) [reply R/path.R]: Oh, right, I rearranged the documentation and moved `path_expand()`, need to importFrom, thanks!

File: NEWS.md (hunk 1 above)
Comment (jimhester) [reply NEWS.md]: Or both here and the help!

**Comment by jimhester** [comment]
Thanks for the review! Will merge assuming the CI passes.

File: NEWS.md [hunk 7]
```diff
@@ -2,6 +2,14 @@
 
 ## Breaking changes
 
+* `path_expand()` and `path_home()` now use `USERPROFILE` or
+  `HOMEDRIVE`/`HOMEPATH` as the user home directory on Windows. This differs
+  from the definition used in `path.expand()` but is consistent with
+  definitions from other programming environments such as python and rust. This
+  is also more compatible with external tools such as git and ssh, both of
+  which put user-level files in `USERPROFILE` by default. To mimic R's (and
+  previous) behavior use `Sys.setenv("R_FS_HOME" = Sys.getenv("R_USER"))`
```
Comment (hadley) [inline NEWS.md]: Making this an environment variable makes me nervous

**Pull request merged**